### ISSUE TITLE
Profile and/or Event questions

### ISSUE THREAD

**Issue by wbushey** [campuscodefest/ccf#98]
I don't know how useful this would be for other orgs, but in the past, Open Twin Cities has relied a lot on being able to ask participants questions while they were RSVPing for an event (on Eventbrite and Meetup). These questions have been tremendously helpful in understanding who is coming to events, who we should reach out to, what logistical things we should know about (e.g. dietary restrictions), who wants to sign up for newsletters/post event activities, and much more.

I'd love to have this kind of feature in CoEventer, and I think it could be even more useful in CoEventer than in Eventbrite, since responses could be tied to a profiles, allowing answers to persist across events and be used in non-event contexts (e.g. somebody creates an Angular project outside of an event and is looking for collaborators, we could look in CoEventer for people who have indicated that they know Angular).

Does this sound like a good idea for CoEventer?

**Comment by pete2786** [comment]
I do like the idea. My biggest concern would be scope. If we are building a full out survey tool, then we are going to far. If it could be a simple model with CoEventer wide questions that you could enable and maybe  some limited custom question objects, that could work.

**Comment by wbushey** [comment]
I agree there's a big risk of scope blowup on this, and I think a first step should be to see if there's a gem that already does something like this.

**Comment by pete2786** [comment]
@wbushey Sure, it might also be a great opportunity to integrate with another provider. For example, Form Stack or a simple Google doc integration.

**Comment by mjhoy** [comment]
just discussed this with @wbushey and we were thinking about the following implementation:

- there exist a global set of questions for users (profile questions more or less) that any organization can select to be part of their event registration form

- a user must fill out answers to selected questions when registering

- those answers get recorded and are visible in aggregate to the administrator per event

- if a user has previously filled out a question, the answer is pre-populated to it in the registration form, effectively letting them update their answer if they want to

there are a definitely details to work out here. i'm interested in writing up a draft of the implementation if you think it could work.

**Comment by pete2786** [comment]
I think that would probably be close to a good approach for an MVP feature. A couple of questions...

- Is the global session set static/defined at app boot or is it something we would add to the app admin interface?
- What would the supported question types be, I feel like this would be where a lot of complexity would come in for anything besides more or less free form/text. Although, we could store the answer as text and kind of 'hard core questions' and the options for response to keep it simple.
- Would a previously filled out question go across organizations, events? Do I answer these once or is the answer unique per event?

**Comment by wbushey** [comment]
Adding some notes - we talked a bit about personas related to surveying, and what those personas want. These lists are in no particular order, and don't reflect what an MVP should have.

**A CoEventer User Wants**

- Not to be inconvenienced
- Not to re-enter data
- Not to be confused
- Not to be prompted with too many questions

**A CoEventer Admin Wants**

- Ease of adding questions
- Ease of summary visualizations
- Ability to download data
- Users to be able to update answers
- Not to fuss w/ development implementation
- Filtering on Answers
- On Site with Multiple Orgs, each having their own questions
- Data about organization members in one place

**Comment by wbushey** [comment]
We also talked a bit about relying more on an existing survey platform, similar to what @pete2786 [suggested last week](https://github.com/campuscodefest/ccf/issues/98#issuecomment-184902671). We didn't have time to investigate what is already out there, and I would like to research a bit around what, if anything, can be accomplished with embedding or APIing something from Form Stack/Google Docs/SurveyMonkey/etc...

**Comment by mjhoy** [comment]
to your questions @pete2786,

1. i think the goal is to move eventually to have an admin interface for adding questions, but it might be quick to start without one (and simply have a defined set of ruby objects to begin with)
2. i think everything should be text values, and perhaps a 'form_type' attribute that can interpret and output the question as a select list or checkboxes
3. i believe we were thinking that previously answered questions would go across organizations and events, however we were not sure whether updated answers should update everywhere -- in other words, whether an answer meant a new row in the database associated to the event, or if it just updated a 'user_answers' row for any event that user was a part of
